product_default_image/models/res_company.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, tools
+from odoo.modules.module import get_module_resource
+
+from ..image_constants import TYPES
+
+
+class ResCompany(models.Model):
+
+    _name = 'res.company'
+    _description = 'ResCompany'
+    _inherit = ['res.company', 'abstract.product.image']
+
+    product_image_target = fields.Selection(
+        string='Default Product Image',
+        selection=[
+            ('none', 'No Default Image'),
+            ('global', 'Global Product Image'),
+            ('category', "Category's Image"),
+            ('global_category', 'Global and Category'),
+        ],
+        required=True,
+        default='none',
+        help='Determines the type of default images to use. '
+             'Changing this field will set all empty product images '
+             'to the target you specify. Products with a default image '
+             'from the previous target are changed to your new target '
+             'as well.\n\n'
+             'No Default Image: Deletes images.\n\n'
+             'Global: Set to Global Product Image.\n\n'
+             'Category: Set products to their category image.\n\n'
+             'Global and Category: Set products to their category '
+             'image if the category has an image. Otherwise set it to '
+             'the Global Product Image',
+    )
+    product_image = fields.Binary(
+        string='Global Product Image',
+        default=lambda s: s._default_product_image(),
+        help='Use as the global image for all product default images. '
+             'Limited to 1024x1024.',
+    )
+
+    @api.model
+    def _default_product_image(self):
+        image_path = get_module_resource(
+            'product_default_image', 'static/src/img', 'glob_prod_img.png'
+        )
+        return tools.image_resize_image_big(
+            open(image_path, 'rb').read().encode('base64')
```

Comment:
Bad - This file handler is left open.
```
with open(image_path, 'rb') as handler:
    image_data = handler.read()
return tools...
```

You could technically return from inside the with statement, but the handler wouldn't be closed until the end of the session.